logic/tasks/StandardTask.cpp

```diff
@@ -0,0 +1,130 @@
+// Licensed under the Apache-2.0 license. See README.md for details.
```

Comment:
This file should be turned into a task factory for standard tasks, or maybe a bunch of static Something::create methods.